common_utils.py

```diff
@@ -221,21 +221,24 @@ def dragMoveEvent(self, event):
     def dropEvent(self, event):
         data = event.mimeData()
         mime = "text/uri-list"
-        url = str(data.data(mime))
-        path = urlparse(url).path.strip()
-        scheme = urlparse(url).scheme
+        url = urlparse(str(data.data(mime), encoding='utf-8').rstrip())
```

Comment:
What is the sample string, that `data.data(mime)` is returning?

I wonder if `.strip()` can be used instead of `.rstrip()` to be more resilient to unusual strings.

---

Changes in this file look like general bug fixes and should go into a standalone PR with an explanation of a problem (all inputs & outputs) and a solution.

Reply:
In the case of dropping a file, the path contains `\r\n` on the end, at least on Linux.  It looks like it breaks urlparser.  I don't see any added value to `.strip()`, since this is generated by the OS.  Why would it inject something at the beginning?

Reply:
I see now. Thank you.

In fact, extra stripping code won't be necessary anymore, when you'll add a setting (on reader class-level) if they want file chooser dialog or a textarea with exported annotations.